archimedes/registry.py

```diff
@@ -0,0 +1,192 @@
+        to deal with their IDs and titles. The registry is saved in the root folder of Archimedes,
+        with name .registry. The content of the registry is as following:
+        [
+            "1": {
+                    'id': 'Search:_pull_request:false',
+                    'title': 'Search:_pull_request:false',
+                    'type': 'search',
+                    'version': 1
+            },
+            "2": {
+                    'id': '8539ada0-9960-11e8-8771-a349686d998a',
+                    'title': 'dockerhub',
+                    'type': 'index-pattern',
+                    'version': 1
+            },
+            ...
+        ]
+
+        :param root_path: path where the registry will be stored
+        """
+        self.path = os.path.join(root_path, REGISTRY_NAME)
+
+        if not os.path.exists(self.path):
+            self.__create_registry()
+
+    def find(self, alias=None, obj_type=None):
+        """Return the information in the registry. If `alias` and `obj_type` are null, it returns the
+        content of all the registry. Otherwise, it returns the information related to the single `alias` or
+        the aliases with the given `obj_type`.
+
+        :param alias: target alias
+        :param obj_type: target object type
+
+        :returns the information related to the alias
+        """
+        registry = load_json(self.path)
+
+        if alias:
+            if alias not in registry:
+                cause = "Alias %s not found in registry" % alias
+                logger.error(cause)
+                raise NotFoundError(cause=cause)
+
+            meta = MetaObj.create_from_registry(registry[alias])
+            yield alias, meta
+
+        elif obj_type:
+            for alias in registry.keys():
+                if registry[alias]['type'] == obj_type:
+                    meta = MetaObj.create_from_registry(registry[alias])
+                    yield alias, meta
+        else:
+            for alias in registry.keys():
+                meta = MetaObj.create_from_registry(registry[alias])
+                yield alias, meta
+
+    def delete(self, alias=None):
+        """Delete the registry content. If `alias` if not null, it deletes only the information
+        related to that alias.
+
+        :param alias: target alias
+        """
+        registry = load_json(self.path)
+
+        if not alias:
+            registry.clear()
+
+            self.__save_registry(registry)
+            logger.info("Registry cleared")
+            return
+
+        if alias not in registry:
+            cause = "Alias %s not found in registry" % alias
+            logger.error(cause)
+            raise NotFoundError(cause=cause)
+
+        registry.pop(alias)
+
+        self.__save_registry(registry)
+        logger.info("Alias %s deleted", alias)
+
+    def add(self, obj, force=False):
+        """Add the meta information of an object to the registry.
+
+        :param obj: the object to save to the registry
```

Comment:
When we talk about objects, we should be really specific. Everything in Python are objects, so when someone reads the documentation, he might think it is referring to other kind of objects and not to those you have in mind.